### ISSUE TITLE
absolute pdn offset

### ISSUE THREAD

**Issue by kareefardi** [The-OpenROAD-Project/OpenLane#1213]
### Prompt
A clear and concise description of what shortcoming you feel OpenLane has.

PDN's offset is measured from the core origin. This is the way openroad api is defined. During developing a design core margin can change frequently resulting in a new PDN interface the given same PDN offset

### Proposal

Using the core margin, core site dimensions, a fixed PDN offset, an offset relative to the core can be calculated at run time to maintain a "fixed" PDN offset. This can ease development of designs and maintenance of new versions of designs

**Comment by maliberty** [comment]
Why would you want PDN to be fixed if the core area is moving underneath it?  You may wind up with a poor grid that doesn't cover the core area.

**Comment by maliberty** [comment]
It would connect to the straps only over the overlapping area (in the extreme they would be unconnected if there was no overlap).  What is the value in having the PDN fixed in space?

**Comment by kareefardi** [comment]
In some cases a given macro is already integrated and it would be hard to change the routing of the parent design

**Comment by maliberty** [comment]
Why change the core margin in that case?

**Comment by maliberty** [comment]
Do you mean make the margin smaller?  Increasing the margin would increase congestion.  I'm not sure why you don't just set it to a minimal value to start with.

**Comment by kareefardi** [comment]
Yes that's what I meant. If the old was big, decreasing it would help. Having a minimal value.
I don't know really know on the top of my head why we aren't using always a minimal margin. I think the relation between the margin, number of pins, congestion, etc.. isn't always directly correlated. I recall that I once had to increase the margin in order to get it to cleanly route.

**Comment by maliberty** [comment]
>  I recall that I once had to increase the margin in order to get it to cleanly route.
That sounds more like a bug.  I think using a minimal value would be better than fixing the pdn grid.
